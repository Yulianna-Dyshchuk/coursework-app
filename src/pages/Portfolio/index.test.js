import React from "react";
import { shallow } from "enzyme";
import { Portfolio } from "./index";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({ pathname: "localhost:3000/" })
}));

describe("Portfolio component", () => {
  it("should render correctly without props", () => {
    const component = shallow(<Portfolio />);

    expect(component).toMatchSnapshot();
  });

  it("should render correctly with props", () => {
    const component = shallow(<Portfolio path="/test" />);

    expect(component).toMatchSnapshot();
  });
});
