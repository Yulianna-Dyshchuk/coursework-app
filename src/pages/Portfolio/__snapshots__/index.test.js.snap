// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio component should render correctly with props 1`] = `
<main>
  <Switch>
    <Route
      exact={true}
      path="/test/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/test/:category"
      render={[Function]}
    />
  </Switch>
</main>
`;

exports[`Portfolio component should render correctly without props 1`] = `
<main>
  <Switch>
    <Route
      exact={true}
      path="undefined/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="undefined/:category"
      render={[Function]}
    />
  </Switch>
</main>
`;
